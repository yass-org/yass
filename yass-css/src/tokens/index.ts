import color from './default/color.json'
import elevation from './default/elevation.json'
import fontWeight from './default/font-weight.json'
import opacity from './default/opacity.json'
import scale from './default/scale.json'

import css from './css.json'

const defaults = [
  ...color,
  ...elevation,
  ...fontWeight,
  ...opacity,
  ...scale,
]

export {
  defaults,
  css,
}