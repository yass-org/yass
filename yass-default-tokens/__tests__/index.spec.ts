import tokens from '../src/index'

describe('yass-default-tokens', () => {
  it('base tokens match snapshot', () => {
    expect(tokens.base).toMatchSnapshot()
  })

  it('component tokens match snapshot', () => {
    expect(tokens.components).toMatchSnapshot()
  })

  it('utility classes match snapshot', () => {
    expect(tokens.utility).toMatchSnapshot()
  })
})

