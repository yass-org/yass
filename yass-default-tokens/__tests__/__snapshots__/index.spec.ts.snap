// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yass-default-tokens base tokens match snapshot 1`] = `
{
  "color": {
    "properties": [
      "background-color",
      "border-bottom-color",
      "border-color",
      "border-left-color",
      "border-right-color",
      "border-top-color",
      "color",
    ],
    "tokens": {
      "blue-100": "hsl(220, 80%, 95%)",
      "blue-200": "hsl(220, 80%, 85%)",
      "blue-300": "hsl(220, 80%, 75%)",
      "blue-400": "hsl(220, 80%, 65%)",
      "blue-50": "hsl(220, 80%, 97.5%)",
      "blue-500": "hsl(220, 80%, 50%)",
      "blue-600": "hsl(220, 80%, 40%)",
      "blue-700": "hsl(220, 80%, 35%)",
      "blue-800": "hsl(220, 80%, 30%)",
      "blue-900": "hsl(220, 80%, 20%)",
      "green-100": "hsl(135, 80%, 95%)",
      "green-200": "hsl(135, 80%, 85%)",
      "green-300": "hsl(135, 80%, 75%)",
      "green-400": "hsl(135, 80%, 45%)",
      "green-50": "hsl(135, 80%, 97.5%)",
      "green-500": "hsl(135, 80%, 35%)",
      "green-600": "hsl(135, 80%, 30%)",
      "green-700": "hsl(135, 80%, 25%)",
      "green-800": "hsl(135, 80%, 20%)",
      "green-900": "hsl(135, 80%, 15%)",
      "neutral-100": "hsl(220, 25%, 95%)",
      "neutral-200": "hsl(220, 25%, 85%)",
      "neutral-300": "hsl(220, 25%, 75%)",
      "neutral-400": "hsl(220, 25%, 65%)",
      "neutral-50": "hsl(220, 25%, 97.5%)",
      "neutral-500": "hsl(220, 25%, 50%)",
      "neutral-600": "hsl(220, 25%, 40%)",
      "neutral-700": "hsl(220, 25%, 30%)",
      "neutral-800": "hsl(220, 25%, 20%)",
      "neutral-900": "hsl(220, 25%, 15%)",
      "orange-100": "hsl(25, 100%, 95%)",
      "orange-200": "hsl(25, 100%, 85%)",
      "orange-300": "hsl(25, 100%, 75%)",
      "orange-400": "hsl(25, 100%, 60%)",
      "orange-50": "hsl(25, 100%, 97.5%)",
      "orange-500": "hsl(25, 100%, 50%)",
      "orange-600": "hsl(25, 100%, 40%)",
      "orange-700": "hsl(25, 100%, 35%)",
      "orange-800": "hsl(25, 100%, 25%)",
      "orange-900": "hsl(25, 100%, 20%)",
      "pink-100": "hsl(320, 80%, 95%)",
      "pink-200": "hsl(320, 80%, 85%)",
      "pink-300": "hsl(320, 80%, 75%)",
      "pink-400": "hsl(320, 80%, 65%)",
      "pink-50": "hsl(320, 80%, 97.5%)",
      "pink-500": "hsl(320, 80%, 50%)",
      "pink-600": "hsl(320, 80%, 40%)",
      "pink-700": "hsl(320, 80%, 35%)",
      "pink-800": "hsl(320, 80%, 30%)",
      "pink-900": "hsl(320, 80%, 20%)",
      "purple-100": "hsl(280, 80%, 95%)",
      "purple-200": "hsl(280, 80%, 85%)",
      "purple-300": "hsl(280, 80%, 75%)",
      "purple-400": "hsl(280, 80%, 65%)",
      "purple-50": "hsl(280, 80%, 97.5%)",
      "purple-500": "hsl(280, 80%, 50%)",
      "purple-600": "hsl(280, 80%, 40%)",
      "purple-700": "hsl(280, 80%, 35%)",
      "purple-800": "hsl(280, 80%, 30%)",
      "purple-900": "hsl(280, 80%, 20%)",
      "red-100": "hsl(0, 100%, 95%)",
      "red-200": "hsl(0, 100%, 85%)",
      "red-300": "hsl(0, 100%, 75%)",
      "red-400": "hsl(0, 100%, 65%)",
      "red-50": "hsl(0, 100%, 97.5%)",
      "red-500": "hsl(0, 100%, 35%)",
      "red-600": "hsl(0, 100%, 30%)",
      "red-700": "hsl(0, 100%, 25%)",
      "red-800": "hsl(0, 100%, 20%)",
      "red-900": "hsl(0, 100%, 15%)",
      "teal-100": "hsl(180, 80%, 95%)",
      "teal-200": "hsl(180, 80%, 85%)",
      "teal-300": "hsl(180, 80%, 75%)",
      "teal-400": "hsl(180, 80%, 45%)",
      "teal-50": "hsl(180, 80%, 97.5%)",
      "teal-500": "hsl(180, 80%, 35%)",
      "teal-600": "hsl(180, 80%, 30%)",
      "teal-700": "hsl(180, 80%, 25%)",
      "teal-800": "hsl(180, 80%, 20%)",
      "teal-900": "hsl(180, 80%, 15%)",
      "yellow-100": "hsl(50, 100%, 95%)",
      "yellow-200": "hsl(50, 100%, 85%)",
      "yellow-300": "hsl(50, 100%, 75%)",
      "yellow-400": "hsl(50, 100%, 60%)",
      "yellow-50": "hsl(50, 100%, 97.5%)",
      "yellow-500": "hsl(50, 100%, 50%)",
      "yellow-600": "hsl(50, 100%, 40%)",
      "yellow-700": "hsl(50, 100%, 35%)",
      "yellow-800": "hsl(50, 100%, 25%)",
      "yellow-900": "hsl(50, 100%, 20%)",
    },
  },
  "elevation": {
    "properties": [
      "z-index",
    ],
    "tokens": {
      "0": "0",
      "100": "100",
      "200": "200",
      "300": "300",
      "400": "400",
      "500": "500",
      "600": "600",
      "700": "700",
      "800": "800",
      "900": "900",
    },
  },
  "fontWeight": {
    "properties": [
      "font-weight",
    ],
    "tokens": {
      "100": "100",
      "200": "200",
      "300": "300",
      "400": "400",
      "500": "500",
      "600": "600",
      "700": "700",
      "800": "800",
      "900": "900",
    },
  },
  "opacity": {
    "properties": [
      "opacity",
    ],
    "tokens": {
      "0": "0%",
      "100": "10%",
      "1000": "100%",
      "200": "20%",
      "300": "30%",
      "400": "40%",
      "500": "50%",
      "600": "60%",
      "700": "70%",
      "800": "80%",
      "900": "90%",
    },
  },
  "scale": {
    "properties": [
      "-webkit-margin-after",
      "-webkit-margin-before",
      "-webkit-padding-after",
      "-webkit-padding-before",
      "-webkit-shape-margin",
      "border-bottom-left-radius",
      "border-bottom-right-radius",
      "border-bottom-width",
      "border-left-width",
      "border-radius",
      "border-right-width",
      "border-top-left-radius",
      "border-top-right-radius",
      "border-top-width",
      "border-width",
      "column-gap",
      "font-size",
      "gap",
      "height",
      "line-height",
      "margin-block-end",
      "margin-block-start",
      "margin-block",
      "margin-bottom",
      "margin-inline-end",
      "margin-inline-start",
      "margin-inline",
      "margin-left",
      "margin-right",
      "margin-top",
      "margin",
      "max-height",
      "max-width",
      "min-height",
      "min-width",
      "padding-block-end",
      "padding-block-start",
      "padding-block",
      "padding-bottom",
      "padding-inline-end",
      "padding-inline-start",
      "padding-inline",
      "padding-left",
      "padding-right",
      "padding-top",
      "padding",
      "row-gap",
      "scroll-margin-bottom",
      "scroll-margin-left",
      "scroll-margin-right",
      "scroll-margin-top",
      "scroll-margin",
      "scroll-padding-block-end",
      "scroll-padding-block-start",
      "scroll-padding-block",
      "scroll-padding-bottom",
      "scroll-padding-inline-end",
      "scroll-padding-inline-start",
      "scroll-padding-inline",
      "scroll-padding-left",
      "scroll-padding-right",
      "scroll-padding-top",
      "scroll-padding",
      "scroll-snap-inline-block",
      "scroll-snap-margin-block-end",
      "scroll-snap-margin-block-start",
      "scroll-snap-margin-block",
      "scroll-snap-margin-bottom",
      "scroll-snap-margin-inline-end",
      "scroll-snap-margin-inline-start",
      "scroll-snap-margin-left",
      "scroll-snap-margin-right",
      "scroll-snap-margin-top",
      "shape-margin",
      "width",
      "top",
      "right",
      "bottom",
      "left",
    ],
    "tokens": {
      "0": "0",
      "1": "0.0625rem",
      "10": "1rem",
      "100": "10rem",
      "15": "1.5rem",
      "150": "15rem",
      "2": "0.125rem",
      "20": "2rem",
      "200": "20rem",
      "25": "2.5rem",
      "250": "25rem",
      "3": "0.25rem",
      "30": "3rem",
      "300": "30rem",
      "35": "3.5rem",
      "350": "35rem",
      "4": "0.375rem",
      "40": "4rem",
      "400": "40rem",
      "45": "4.5rem",
      "450": "45rem",
      "5": "0.5rem",
      "50": "5rem",
      "500": "50rem",
      "55": "5.5rem",
      "550": "55rem",
      "6": "0.625rem",
      "60": "6rem",
      "600": "60rem",
      "65": "6.5rem",
      "650": "65rem",
      "7": "0.75rem",
      "70": "7rem",
      "700": "70rem",
      "75": "7.5rem",
      "750": "75rem",
      "8": "0.875rem",
      "80": "8rem",
      "800": "80rem",
      "85": "8.5rem",
      "850": "85rem",
      "9": "0.9375rem",
      "90": "9rem",
      "900": "90rem",
      "95": "9.5rem",
    },
  },
}
`;

exports[`yass-default-tokens component tokens match snapshot 1`] = `
{
  "__esModule": true,
  "box": [
    {
      "properties": [
        "background-color",
        "border-bottom-color",
        "border-color",
        "border-left-color",
        "border-right-color",
        "border-top-color",
      ],
      "tokens": {
        "danger": "hsl(0, 100%, 35%)",
        "danger-active": "hsl(0, 100%, 30%)",
        "danger-hover": "hsl(0, 100%, 65%)",
        "link": "hsl(220, 80%, 50%)",
        "link-active": "hsl(220, 80%, 40%)",
        "link-hover": "hsl(220, 80%, 65%)",
        "link-visited": "hsl(280, 80%, 50%)",
        "neutral": "hsl(220, 25%, 40%)",
        "neutral-active": "hsl(220, 25%, 30%)",
        "neutral-hover": "hsl(220, 25%, 50%)",
        "primary": "hsl(220, 80%, 50%)",
        "primary-active": "hsl(220, 80%, 35%)",
        "primary-hover": "hsl(220, 80%, 75%)",
        "subtle": "hsl(220, 25%, 75%)",
        "subtle-active": "hsl(220, 25%, 65%)",
        "subtle-hover": "hsl(220, 25%, 85%)",
        "success": "hsl(135, 80%, 35%)",
        "success-active": "hsl(135, 80%, 30%)",
        "success-hover": "hsl(135, 80%, 45%)",
        "warning": "hsl(25, 100%, 50%)",
        "warning-active": "hsl(25, 100%, 40%)",
        "warning-hover": "hsl(25, 100%, 60%)",
      },
    },
    {
      "properties": [
        "z-index",
      ],
      "tokens": {
        "default": "100",
        "overlay": "900",
        "raised": "500",
        "sunken": "0",
      },
    },
  ],
  "button": [
    {
      "properties": [
        "padding-inline",
      ],
      "tokens": {
        "button-large": "2rem",
        "button-medium": "0.875rem",
        "button-small": "0.5rem",
      },
    },
    {
      "properties": [
        "padding-block",
      ],
      "tokens": {
        "button-large": "1rem",
        "button-medium": "0.5rem",
        "button-small": "0.125rem",
      },
    },
    {
      "properties": [
        "font-size",
      ],
      "tokens": {
        "button-large": "1rem",
        "button-medium": "0.875rem",
        "button-small": "0.875rem",
      },
    },
  ],
  "grid": [
    {
      "properties": [
        "grid-template-columns",
      ],
      "tokens": {
        "page-layout": "1fr 4fr 1fr",
        "three-column-layout": "repeat(3, 1fr)",
        "two-column-layout": "repeat(2, 1fr)",
      },
    },
  ],
  "text": [
    {
      "properties": [
        "font-weight",
      ],
      "tokens": {
        "bold": "700",
        "light": "200",
        "normal": "400",
      },
    },
    {
      "properties": [
        "color",
      ],
      "tokens": {
        "danger": "hsl(220, 25%, 95%)",
        "danger-active": "hsl(220, 25%, 85%)",
        "danger-hover": "hsl(220, 25%, 97.5%)",
        "neutral": "hsl(220, 25%, 95%)",
        "neutral-active": "hsl(220, 25%, 85%)",
        "neutral-hover": "hsl(220, 25%, 97.5%)",
        "primary": "hsl(220, 25%, 95%)",
        "primary-active": "hsl(220, 25%, 85%)",
        "primary-hover": "hsl(220, 25%, 97.5%)",
        "subtle": "hsl(220, 25%, 30%)",
        "subtle-active": "hsl(220, 25%, 20%)",
        "subtle-hover": "hsl(220, 25%, 40%)",
        "success": "hsl(220, 25%, 95%)",
        "success-active": "hsl(220, 25%, 85%)",
        "success-hover": "hsl(220, 25%, 97.5%)",
        "text": "hsl(220, 25%, 20%)",
        "warning": "hsl(220, 25%, 95%)",
        "warning-active": "hsl(220, 25%, 85%)",
        "warning-hover": "hsl(220, 25%, 97.5%)",
      },
    },
  ],
}
`;

exports[`yass-default-tokens utility classes match snapshot 1`] = `
[
  {
    "declarations": [
      {
        "property": "position",
        "value": "absolute",
      },
      {
        "property": "width",
        "value": "1px",
      },
      {
        "property": "height",
        "value": "1px",
      },
      {
        "property": "margin",
        "value": "-1px",
      },
      {
        "property": "border",
        "value": "0",
      },
      {
        "property": "padding",
        "value": "0",
      },
      {
        "property": "white-space",
        "value": "no-wrap",
      },
      {
        "property": "clip-path",
        "value": "inset(100%)",
      },
      {
        "property": "clip",
        "value": "rect(0 0 0 0)",
      },
      {
        "property": "overflow",
        "value": "hidden",
      },
    ],
    "name": "visually-hidden",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(0, 100%, 15%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(0, 100%, 15%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-red",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(25, 100%, 20%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-orange",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(50, 100%, 20%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-yellow",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(0, 100%, 15%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-green",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(180, 80%, 15%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-teal",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(220, 80%, 20%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-blue",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(280, 80%, 20%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-purple",
  },
  {
    "declarations": [
      {
        "property": "background-color",
        "value": "hsl(320, 80%, 20%)",
      },
      {
        "property": "opacity",
        "value": "10%",
      },
    ],
    "name": "debug-pink",
  },
]
`;
